wcfsetup/install/files/lib/system/mail/Mail.class.php

```diff
@@ -272,7 +158,14 @@ public function addTo($to) {
 	 * @return	mixed
 	 */
 	public function getTo() {
```

Comment:
The change violates the contract that:
```
$mail = new Mail();
$mail->addTo($x);
// $mail->getTo()[0] === $x; for all $x given !is_array($x)
```

Specifically the `display-name` of the address specification is being stripped. In fact this is one of the reasons I did not implement the getters in #2756: It is impossible to map the methods without a behavioural change. Thus I opted for a highly visible exception instead of a silent change in behaviour. If people absolutely have to use the getters (which I doubt) they would need to rewrite their code anyway and thus can simply use the new email API (which exists since 2 years). If they don't, then #2756 is fine for them and they get to reap the benefits of the new email API without doing anything in the short term.

Reply:
The display name has never been there, as you can see:
https://github.com/WoltLab/WCF/blob/3.0/wcfsetup/install/files/lib/system/mail/Mail.class.php#L283-L285
https://github.com/WoltLab/WCF/blob/3.0/wcfsetup/install/files/lib/system/mail/Mail.class.php#L261
https://github.com/WoltLab/WCF/blob/3.0/wcfsetup/install/files/lib/system/mail/Mail.class.php#L242-L244

Reply:
If a raw string (i.e. no array) is given to `addTo` it is passed through straight. I've seen examples in the wild where a string in `name-addr` format is explicitely given instead of letting `Mail` construct it. See: https://github.com/WoltLab/WCF/commit/b694faf3bfa8a7be7eafcc21568728785ec24c4e#diff-dcefc405b8d94fc948f8ae36d00d2e26L178